src/Shared/PackageManagers/NPMProjectManager.cs

```diff
@@ -148,6 +148,28 @@ public override async Task<bool> PackageExistsAsync(PackageURL purl, bool useCac
             return await CheckJsonCacheForPackage(httpClient, $"{ENV_NPM_API_ENDPOINT}/{packageName}", useCache);
         }
 
+        /// <inheritdoc />
+        public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool useCache = true) // MICHELLE
+        {
+            Logger.Trace("PackageVersionExists {0}", purl?.ToString());
+            if (string.IsNullOrEmpty(purl?.Name))
+            {
+                Logger.Trace("Provided PackageURL was null.");
+                return false;
+            }
+
+            if (purl.Version.IsBlank())
+            {
+                Logger.Trace("Provided PackageURL version was null or blank.");
+                return false;
+            }
+
+            string packageName = purl.HasNamespace() ? $"{purl.GetNamespaceFormatted()}/{purl.Name}/{purl.Version}" : $"{purl.Name}/{purl.Version}";
```

Comment:
nit: not technically the `packageName` I would probably split this up to just the `packageName` declaration and finish the rest of the path in the actual query itself.